dangerfile.js

```diff
@@ -0,0 +1,62 @@
+const execa = require('execa');
+const { fail } = require('danger');
+
+const fromRoot = path => path.replace(`${process.cwd()}/`, '');
+const codeFence = str => `\`\`\`\n${str.trim()}\n\`\`\``;
+
+const tasks = [
+    function prettierCheck() {
+        try {
+            execa.sync('npm', ['run', '--silent', 'prettier:check']);
+        } catch (err) {
+            const { stdout } = err;
+            fail(
+                'The following file(s) were not ' +
+                    'formatted with **prettier**. Make sure to execute `npm run prettier` ' +
+                    `locally prior to committing.\n${codeFence(stdout)}`
+            );
+        }
+    },
+
+    function eslintCheck() {
+        try {
+            execa.sync('npm', ['run', '--silent', 'lint', '--', '-f', 'json']);
```

Comment:
What does this boil down to? `npm run --silent lint -- -f json`?

Reply:
- `npm run --silent` prevents npm from spitting any of its own info to `stdout` (it normally shows the command you ran, and spits info to `stderr` on failures, which we don't want)

- `--` tells `npm` to pass all other arguments past that point directly to the command being run
	> As of npm@2.0.0, you can use custom arguments when executing scripts. The special option -- is used by getopt to delimit the end of the options. npm will pass all the arguments after the -- directly to your script.

- `-f json` is shorthand for `--format json`, which gives us parseable output from the ESLint CLI.